### DISCUSSION TITLE
A thank you note for the HLS team

### DISCUSSION THREAD

**Discussion by pepeiborra** [haskell/haskell-language-server#1169] [General]
I opened my email this morning and found this message of gratitude:

> SUBJECT: kudos on Haskell language editor extension
> I am new to Haskell programming though not new to Haskell, having met Simon Peyton Jones back in 2008 at ICFP Victoria.
> I was doing the exercises when the helper prompted:
```
{
	"owner": "Haskell (h)",
	"code": "refact:Use foldr",
	"severity": 2,
	"message": "Use foldr\nFound:\n  s [] = 0\ns (x : xs) = x + s xs\nWhy not:\n  s xs = foldr (+) 0 xs\n",
	"source": "hlint",
	"startLineNumber": 12,
	"startColumn": 1,
	"endLineNumber": 13,
	"endColumn": 20
}
```
> Indeed, why not foldr!
> I thought to myself, who of the gods wrote the code. github tells me that there are in fact 108 of you!
> I can’t write to everyone to thank them for the valiant effort so I randomly picked you.
> Thank you! Feel free to forward my compliments to your comrades.
> Yih Lerh
> — an old hacker (in the 'code monkey' sense, not in the ‘guru’ sense)

**Comment by jneira** [comment]
Thanks for sharing it, those messages are the best motivation anyone could have

**Comment by googleson78** [comment]
I also want to express huge gratitude! One year ago, I would have never imagined it's possible to so quickly make so much progress in the UX area of Haskell usage!
